src/flask_container_scaffold/app_scaffold.py

```diff
@@ -2,76 +2,99 @@
 
 from flask import Flask
 
-from flask_container_scaffold.util import load_yaml
+from flask_container_scaffold.app_configurator import AppConfigurator
 
 
 class AppScaffold(object):
 
     def __init__(self, app=None,
-                 name=__name__, config=None):
+                 name=__name__, config=None,
+                 settings_required=False,
+                 instance_path=None,
+                 instance_relative_config=True):
         """
         This class provides a way to dynamically configure a Flask application.
 
-        :param app: An existing Flask application, if passed, otherwise we will
-                    create a new one
+        :param app obj: An existing Flask application, if passed, otherwise we
+                    will create a new one
         :param name: The name of the application, defaults to __name__.
```

Comment:
(Minor) May want to specify 'str' for this, since the other all have a type now? (I find these hints very useful personally, thanks for adding them!)